servicetalk-buffer-api/src/main/java/io/servicetalk/buffer/api/CharSequences.java

```diff
@@ -55,6 +55,17 @@ private CharSequences() {
         // No instances
     }
 
+    /**
+     * Create a new {@link CharSequence} from the specified {@code input}, supporting only 8-bit ASCII characters, and
+     * with a case-insensitive {@code hashCode}.
+     *
+     * @param input a character sequence containing only 8-bit ASCII characters.
+     * @return a {@link CharSequence}
+     */
```

Comment:
add `@Deprecated` annotation and javaodc?

Reply:
I don't want to add deprecation because the only fix for user code is to add an explicit upcast to CharSequence which will be redundant once the String method is actually removed. We should investigate solutions to add an invisible bridge method for the String overload so that source compiles to the CharSequence method but binary compatibility is maintained.